src/dataworker/Dataworker.ts

```diff
@@ -106,13 +110,42 @@ export class Dataworker {
     };
   }
 
-  async buildSlowRelayRoot(bundleBlockNumbers: BundleEvaluationBlockNumbers) {
-    this._loadData();
-
-    // For each destination chain ID key in unfilledDeposits
-    //     Order by fillAmountRemaining
-    //     Make Leaf for fill
-    // Construct root
+  async buildSlowRelayRoot(bundleBlockNumbers: BundleEvaluationBlockNumbers): Promise<MerkleTree<RelayData>> | null {
+    const { unfilledDeposits } = this._loadData();
+    // TODO: Use `bundleBlockNumbers` to decide how to filter which blocks to keep in `unfilledDeposits`.
+
+    if (Object.keys(unfilledDeposits).length === 0) return null;
+
+    const leaves: RelayData[] = [];
+    Object.keys(unfilledDeposits).forEach((destinationChainId) => {
+      leaves.push(
+        ...unfilledDeposits[destinationChainId].map((deposit: UnfilledDeposit): RelayData => {
+          return {
+            depositor: deposit.deposit.depositor,
+            recipient: deposit.deposit.recipient,
+            destinationToken: deposit.deposit.depositor,
+            amount: deposit.deposit.amount,
+            originChainId: deposit.deposit.originChainId,
+            destinationChainId: deposit.deposit.destinationChainId,
+            realizedLpFeePct: deposit.deposit.realizedLpFeePct,
+            relayerFeePct: deposit.deposit.relayerFeePct,
+            depositId: deposit.deposit.depositId,
+          };
+        })
+      );
+    });
```

Comment:
You can take or leave this suggestion. I just thought the nested map + .flat was a little simpler, but I could see the other way too.
```suggestion
    const leaves: RelayData[] = Object.values(unfilledDeposits).map((deposits) => deposits.map((deposit: UnfilledDeposit): RelayData => ({
            depositor: deposit.deposit.depositor,
            recipient: deposit.deposit.recipient,
            destinationToken: deposit.deposit.depositor,
            amount: deposit.deposit.amount,
            originChainId: deposit.deposit.originChainId,
            destinationChainId: deposit.deposit.destinationChainId,
            realizedLpFeePct: deposit.deposit.realizedLpFeePct,
            relayerFeePct: deposit.deposit.relayerFeePct,
            depositId: deposit.deposit.depositId,
        })
      ).flat();
```

Reply:
I really dont like this syntax. @nicholaspai and I spoke about doing some casting between `Deposit` and `RelayData` but there are ts issues with this.

 Ideally, we want the ability to do the following:
1) extract all the keys within `RelayData`
2) for each key assign a value in the `relayData` from the given key from `Deposit`. This basically makes this spread, assign and cast operation programatic, rather than manual the way you have it now. Only issue is I have not found a predictable way to extract keys from ts interfaces. I have tried using this: https://github.com/kimamula/ts-transformer-keys but it was not consistent. 

Another option would be to have a const in the interfaces file that defines all the keys you are expecting within the `RelayData`. this can then be traversed and spread from the deposit object.

Reply:
@chrismaree what is your suggestion for replacing this syntax? I think this is type safe as this will fail if the manual creation of the RelayData forgets a key